pom.xml

```diff
@@ -1793,6 +1811,26 @@
                                 <compilerArgs combine.self="append">
                                     <compilerArg>-parameters</compilerArg>
                                 </compilerArgs>
+                                <testExcludes>
+                                    <testExclude>**/*NoParametersCompilerFlag*</testExclude>
+                                </testExcludes>
+                            </configuration>
+                        </execution>
+                        <execution>
+                            <id>noparameters-testCompile</id>
+                            <phase>test-compile</phase>
+                            <goals>
+                                <goal>testCompile</goal>
+                            </goals>
+                            <configuration>
+                                <skip>${maven.compiler.testSources.noParamterCompilation.skip}</skip>
+                                <fork>true</fork>
+                                <proc>none</proc>
+                                <executable>${java-version.test.compiler}</executable>
+                                <compilerArgs combine.self="override" />
+                                <testIncludes>
+                                    <testInclude>**/*NoParametersCompilerFlag*</testInclude>
+                                </testIncludes>
```

Comment:
Is there any way you could use the directory name instead, e.g. `src/test/java17-noparameters/**`? So that we don't need tests to be named a certain way...

Reply:
That's a bummer :/

> Because of how the filter works (ignores the source location), that was why I was asking if there's a point to keeping sources separately.

You're right, if we can't differentiate between source directories in the maven-compiler-plugin, then it doesn't make sense to use a different directory for tests compiled without `-parameters`.

Still, filtering by class name feels very fragile, as class renaming is not unusual is and is usually expected to have to side-effect... Maybe you could use the package name in the filter instead? Not sure that would be much better, but perhaps a little bit...

At least the test will fail if compilation used `-parameters`, so I guess there's that...